sim/core/unit.go

```diff
@@ -686,3 +693,16 @@ func (unit *Unit) GetTotalAvoidanceChance(atkTable *AttackTable) float64 {
 	block := unit.GetTotalBlockChanceAsDefender(atkTable)
 	return miss + dodge + parry + block
 }
+
+func (unit *Unit) MultiplyMovementSpeed(sim *Simulation, amount float64) {
+	unit.PseudoStats.MovementSpeedMultiplier *= amount
+}
```

Comment:
This should check for any movement currently happening and updated timers/durations based on new speed